pkg/cri/server/sandbox_run.go

```diff
@@ -95,6 +84,40 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		},
 	)
 
+	// cleanupErr records the last error created by the critical cleanup in defer functions,
+	// like CNI teardown and stop running sandbox task.
+	// So when the cleanup is not completed for some reason, the CRI-plugin will remain the sandbox
+	// in the not-ready state, which can be cleanup by kubelet syncPod.
+	var cleanupErr error
+
+	defer func() {
+		if cleanupErr != nil {
```

Comment:
hmm, I believe my current code works in that way?

If `c.sandboxNameIndex.Reserve` fails, this function returns error `failed to reserve sandbox name`. This defer will be no-op because cleanupErr is nil.

Reply:
Oh, maybe you mean move this `defer function` below `c.sandboxNameIndex.Reserve`?

I think both way have equivalent logic. For readability, I make the this "persist sandbox/container if cleanup fails" at the top and put `sandboxNameIndex.Reserve` and `sandboxNameIndex.ReleaseByName` together

Reply:
I moved the rollback downwards.